### PR TITLE
fix #7600 add hookscript parameter to proxmox_kvm

### LINKED ISSUE #7600: add hookscript parameter to proxmox_kvm
### Summary

I would like to add the config `hookscript` parameter which is not exposed via the proxmox_kvm module.

### Issue Type

Feature Idea

### Component Name

proxmox_kvm

### Additional Information

```yaml (paste below)
- name: Create new VM with minimal options and hookscript
  community.general.proxmox_kvm:
    # existing params
    api_user: root@pam
    api_password: secret
    api_host: helldorado
    name: spynal
    node: sabrewulf
    # this param is new:
    hookscript: local:snippets/myhook.py
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by elelay**
##### SUMMARY

Fixes #7600 add hookscript parameter to proxmox_kvm module.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
proxmox_kvm

##### ADDITIONAL INFORMATION

The parameter was present in PROXMOX VE as early as [April 2019](https://forum.proxmox.com/threads/how-to-use-new-hookscript-feature.53388/).
It had been added to the `proxmox.py` module in #245 on May 18, 2020 but not to `proxmox_kvm.py`.
This PR adds it.

### CODE REVIEW

File: changelogs/fragments/7600-proxmox_kvm-hookscript.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - "proxmox_kvm - support the ``hookscript`` parameter (https://github.com/ansible-collections/community.general/issues/7600)"
```
Comment (felixfontein) [inline changelogs/fragments/7600-proxmox_kvm-hookscript.yml:2]:
```suggestion
  - "proxmox_kvm - support the ``hookscript`` parameter (https://github.com/ansible-collections/community.general/issues/7600)."
```

File: plugins/modules/proxmox_kvm.py [hunk 2]
```diff
@@ -196,6 +196,10 @@
       - Used only with clone
     type: bool
     default: true
+  hookscript:
+    description:
+      - Script that will be executed during various steps in the containers lifetime.
+    type: str
```
Comment (felixfontein) [inline plugins/modules/proxmox_kvm.py:202]:
```suggestion
    type: str
    version_added: 8.1.0
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution!

**Comment by felixfontein** [comment]
This needs to be updated when #7540 is merged first, and vice versa.

**Comment by elelay** [comment]
> does the `update` code also need to know about this new option?

I've tested the update and delete modes: at least they work on PVE 8.1. Not sure about a previous version.

**Comment by felixfontein** [comment]
From the code this new option shouldn't have an impact on existing tasks (that don't use the option), so even if update doesn't work for this new option for some versions, it won't break any existing user of the module (unless they start using this option). So let's see whether someone complains that it doesn't work :)

**Comment by felixfontein** [comment]
If nobody objects, I'll merge this at the end of the upcoming week (~around Friday).

**Comment by elelay** [comment]
Thanks for the welcoming and positive feedback!

**Comment by felixfontein** [comment]
@elelay thanks for your contribution!

**Pull request merged**